pcmpl-args.el

```diff
@@ -3389,6 +3390,136 @@ options found in its man page."
 (defalias 'pcomplete/traceroute 'pcmpl-args-pcomplete-on-man)
 (defalias 'pcomplete/wget 'pcmpl-args-pcomplete-on-man)
 
+
+;; Pass completion
+
+(defalias 'pcmpl-args-pass-commands
+  (pcmpl-args-completion-table-with-annotations
+   '(("cp" "Copy password or directory")
+     ("edit" "Insert a new password or edit an existing password")
+     ("find" "List names of passwords inside the tree that match patterns")
+     ("generate" "Generate new password")
+     ("git" "Execute git commands")
+     ("grep" "Searches inside each decrypted password file")
+     ("help" "Show usage message")
+     ("init" "Initialize new password storage")
+     ("insert" "Insert a new password into the password store")
+     ("ls" "List names of passwords")
+     ("mv" "Move password or directory")
+     ("rm" "Remove password or directory")
+     ("show" "Decrypt and print a password")
+     ("version" "Show version information"))))
+
+(defun pcmpl-args-pass-prefix ()
+  (let ((directory (or (getenv "PASSWORD_STORE_DIR")
+                       (expand-file-name "~/.password-store"))))
+    (concat directory "/")))
+
+(defun pcmpl-args-pass-find (&rest find-args)
+  (let* ((prefix (pcmpl-args-pass-prefix))
+         (args `("find" "-L" ,prefix
+                 "(" "-name" ".git*" "-o" "-name" ".gpg-id" ")" "-prune"
+                 "-o" ,@find-args "-print"))
+         (rx (concat "^" (regexp-quote prefix) "\\(.+?\\)\\(?:\\.gpg\\)?$")))
+    (with-temp-buffer
+      (apply #'pcmpl-args-process-file args)
+      (goto-char (point-min))
+      (save-match-data
+        (while (search-forward-regexp rx nil t)
+          (replace-match "\\1")))
+      (beginning-of-line)
+      (let (lines)
+        (while (progn (push (string-trim-right (thing-at-point 'line t)) lines)
+                      (forward-line -1)
+                      (not (bobp))))
+        lines))))
+
+(defun pcmpl-args-pass-keys (args)
+  (let ((rx "^\\(?:[^:]*:\\)\\{9\\}\\([^:]*\\).*?$"))
+    (with-temp-buffer
+      (pcmpl-args-process-file "gpg2" "--list-secret-keys" "--with-colons")
+      (goto-char (point-min))
+      (save-match-data
+        (while (search-forward-regexp rx nil t)
+          (replace-match "\\1")))
+      (forward-line 1)
+      (let (lines)
+        (while (progn (forward-line -1)
+                      (push (string-trim-right (thing-at-point 'line t)) lines)
+                      (not (bobp))))
+        (cl-set-difference lines (cadr (assq '* args)) :test #'string=)))))
+
+(defun pcmpl-args-pass-command-specs (cmd)
```

Comment:
Can you add a new section to the end of the README explaining how to define new pass-through completions?

Reply:
What do you mean by "pass-through" completions?